aisdc/attacks/worst_case_attack.py

```diff
@@ -24,6 +24,7 @@
 logging.basicConfig(level=logging.INFO)
 
 P_THRESH = 0.05
+AUC_FAIL_THRESH = 0.8
```

Comment:
In fact, I'd also maybe consider making the object stateful so that it has its own counter and only fails once a certain condition has been met N times.

The rationale is that it would be better (IMO) to be checking the AUC p-value (significance of AUC values massively depends on the size of the dataset (an AUC of 0.75 is within 3SDs of 0.5 for a test dataset with 20 examples from each class). However, one p-value < (say) 0.05 in 100 reps is not a problem (it'll almost certainly disappear on multiple testing)., so perhaps stopping if it happens 10 times, etc makes more sense.

Similarly for the p-value that comes from the FDIF metric

Reply:
Dear Simon, 

Thanks for the feedback/reviewing this PR. We discussed a few more things in last week's catch-up. I agree that we should make the fail-fast option more generic. Given your feedback and the discussion we had last week, I will make changes. Once all that is done, I will update this PR.